sway-fmt-v2/src/utils/comments.rs

```diff
@@ -0,0 +1,182 @@
+use crate::fmt::{Formatter, FormatterError};
+use std::{cmp::Ordering, collections::BTreeMap, fmt::Write, sync::Arc};
+use sway_parse::token::{lex_commented, Comment, CommentedTokenTree, CommentedTree};
+
+/// Represents a span for the comments in a spesific file
+/// A stripped down version of sway-types::src::Span
+#[derive(PartialEq, Eq, Debug, Clone, Default)]
+pub struct CommentSpan {
+    // The byte position in the string of the start of the span.
+    pub start: usize,
+    // The byte position in the string of the end of the span.
+    pub end: usize,
+}
+
+impl Ord for CommentSpan {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // If the starting position is the same encapsulatig span (i.e, wider one) should come
+        // first
+        match self.start.cmp(&other.start) {
+            Ordering::Equal => self.end.cmp(&other.end),
+            ord => ord,
+        }
+    }
+}
+
+impl PartialOrd for CommentSpan {
+    fn partial_cmp(&self, other: &CommentSpan) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+pub type CommentMap = BTreeMap<CommentSpan, Comment>;
+
+/// Get the CommentedTokenStream and collect the spans -> Comment mapping for the input source
+/// code.
+pub fn construct_comment_map(input: Arc<str>) -> Result<CommentMap, FormatterError> {
+    let mut comment_map = BTreeMap::new();
+
+    // pass the input through lexer
+    let commented_token_stream = lex_commented(&input, 0, input.len(), None)?;
+    let tts = commented_token_stream.token_trees().iter();
+
+    for comment in tts {
+        get_comment_from_token_stream(comment, &mut comment_map);
+    }
+    Ok(comment_map)
+}
+
+/// Get `Comment` from the token stream and insert it with its span to the `CommentMap`.
+/// Handles both the standalone and in-block comments.
+fn get_comment_from_token_stream(
```

Comment:
In Rust, `get` is generally used to refer to the retrieval of a `&` to an element within a datastructure (e.g. `Vec::get`, `HashMap::get`, `BTreeMap::get`, etc). Perhaps a better name for this would `collect_comments_from_token_stream`?